primitive-types/impls/serde/src/serialize.rs

```diff
@@ -11,7 +11,33 @@ use std::fmt;
 
 static CHARS: &[u8] = b"0123456789abcdef";
 
-fn to_hex<'a>(v: &'a mut [u8], bytes: &[u8], skip_leading_zero: bool) -> &'a str {
+/// Serialize given bytes to a 0x-prefixed hex string.
+///
+/// If `skip_leading_zero` initial 0s will not be printed out,
+/// unless the byte string is empty, in which case `0x0` will be returned.
+/// The results are consistent with `serialize_uint` output if the flag is
+/// on and `serialize_raw` if the flag is off.
+pub fn to_hex(bytes: &[u8], skip_leading_zero: bool) -> String {
+	let bytes = if skip_leading_zero {
+		let non_zero = bytes.iter().take_while(|b| **b == 0).count();
+		let bytes = &bytes[non_zero..];
+		if bytes.is_empty() {
+			return "0x0".into()
+		} else {
+			bytes
+		}
+	} else if bytes.is_empty() {
+		return "0x".into()
```

Comment:
```suggestion
		return "0x0".into()
```

Reply:
But empty byte set (not uint) should be encoded as `0x` not `0x0`, `vec![0]` should have `0x00` encoding, one byte = two nibbles (always, since we don't trim zeros).